.github/workflows/algolia/src/syncAlgoliaIndex.ts

```diff
@@ -0,0 +1,130 @@
+import fs from "fs";
+import path from "path";
+import matter from "gray-matter";
+import algoliasearch from "algoliasearch";
+
+type DocsFrontMatter = {
+  content: string;
+  data: {
+    objectId: string;
+    title?: string;
+    description?: string;
+    slug?: string;
+    tags?: Array<string>;
+  };
+};
+
+const getAllFiles = (
+  dirPath,
+  arrayOfFiles: Array<{ inputPath: string; outputPath: string; type: string }>,
+  type
+) => {
+  const files = fs.readdirSync(dirPath);
+
+  arrayOfFiles = arrayOfFiles || [];
+
+  files.forEach(function (file) {
+    if (fs.statSync(dirPath + "/" + file).isDirectory()) {
+      arrayOfFiles = getAllFiles(dirPath + "/" + file, arrayOfFiles, type);
+    } else {
+      const inputPath = path.join(dirPath, "/", file);
+      const outputPath = `.\\output\\${path.join(
+        "./output/",
+        dirPath,
+        "/",
+        file
+      )}`;
+      arrayOfFiles.push({ inputPath, outputPath, type });
+    }
+  });
+
+  return arrayOfFiles;
+};
+
+type AlgoliaRecord = {
+  objectId: any;
```

Comment:
Let’s try using `unknown` instead of `any` where we can. `any` disables all type checks, which may bite us one day somewhere. If you are not planning to use the field, `unknown` will save you from finding the right type just as `any`, but  keep TypeScript on guard.